readthedocs/vcs_support/base.py

```diff
@@ -33,60 +32,12 @@ def __repr__(self):
                                        self.verbose_name)
 
 
-class VCSProject(namedtuple("VCSProject",
-                            "name default_branch working_dir repo_url")):
-
-    """Transient object to encapsulate a projects stuff"""
-
-    pass
-
-
-class BaseCLI(object):
-
-    """Helper class for CLI-heavy classes."""
-
-    log_tmpl = u'VCS[{name}:{ident}]: {args}'
-
-    def __call__(self, *args):
-        return self.run(args)
-
-    def run(self, *args):
-        """:param bits: list of command and args. See `subprocess` docs"""
-        process = subprocess.Popen(args, stdout=subprocess.PIPE,
-                                   stderr=subprocess.PIPE,
-                                   cwd=self.working_dir, shell=False,
-                                   env=self.env)
-        try:
-            log.info(self.log_tmpl.format(ident=basename(self.working_dir),
-                                          name=self.name,
-                                          args=' '.join(args)))
-        except UnicodeDecodeError:
-            # >:x
-            pass
-        stdout, stderr = process.communicate()
-        try:
-            log.info(self.log_tmpl.format(ident=basename(self.working_dir),
-                                          name=self.name,
-                                          args=stdout))
-        except UnicodeDecodeError:
-            # >:x
-            pass
-        return (
-            process.returncode,
-            stdout.decode('utf-8'),
-            stderr.decode('utf-8'))
-
-    @property
-    def env(self):
-        return os.environ.copy()
-
-
-class BaseVCS(BaseCLI):
+class BaseVCS(object):
 
     """
     Base for VCS Classes.
 
-    Built on top of the BaseCLI.
+    VCS commands are ran inside a ``LocalEnvironment``.
```

Comment:
In fact, could be any `Environment` not just `Local`. By default, is `LocalEnvironment`